src/react/hooks/__tests__/useQuery.test.tsx

```diff
@@ -2092,6 +2092,200 @@ describe("useQuery Hook", () => {
       unmount();
       result.current.stopPolling();
     });
+
+    describe("should stop refetching if `shouldRefetchOnPolling` is passed at", () => {
+      beforeEach(() => {
+        jest.useFakeTimers();
+      });
+
+      afterEach(() => {
+        jest.runOnlyPendingTimers();
+        jest.useRealTimers();
+      });
+
+      it("global level", async () => {
+        document.hasFocus = jest.fn().mockImplementation(() => true);
```

Comment:
```suggestion
let shouldRefetch = true;
```

Since this is a more generic API with the callback approach, I'd avoid the reference to `document` here and instead just use a variable that you can toggle on/off for the desired behavior.